src/main/js/helpers/eventline.js

```diff
@@ -0,0 +1,145 @@
+ *
+ * @private
+ * @param {object} scale - d3 scale taking into account the input parameters
+ * @param {object} config - config object derived from input JSON
+ * @param {Array} canvasSVG - d3 object of canvas svg
+ * @param {Function} yAxisPositionHandler - call back to get y-axis position.Functions differ based on Constructs used
+ * @returns {undefined} - returns nothing
+ */
+const translateEventline = (scale, config, canvasSVG, yAxisPositionHandler) => {
+    if (utils.isEmpty(config.eventline)) {
+        return;
+    }
+    const eventlineGroup = canvasSVG
+        .selectAll(`.${styles.eventlineGroup}`)
+        .transition()
+        .call(constants.d3Transition(config.settingsDictionary.transition))
+        .attr(
+            "transform",
+            `translate(${getXAxisXPosition(config)},${yAxisPositionHandler(
+                config
+            )})`
+        );
+    eventlineGroup
+        .selectAll(`.${styles.eventlinePoint}`)
+        .select("g")
+        .attr("transform", function(d) {
+            return `translate(${scale.x(
+                utils.parseDateTime(d.value)
+            )}, 0) scale(${getTransformScale(this)})`;
+        });
+    eventlineGroup
+        .selectAll(`.${styles.eventline}`)
+        .attr("x1", (val) => scale.x(utils.parseDateTime(val.value)))
+        .attr("y1", 0)
+        .attr("x2", (val) => scale.x(utils.parseDateTime(val.value)))
+        .attr("y2", getYAxisHeight(config));
+};
+/**
+ * Creates a eventline for graph. We are not adding
+ * x1, x2, y1, y2 co-ordinates for the straight line since they will be adjusted when
+ * content is loaded.
+ *
+ * @private
+ * @param {object} scale - d3 scale taking into account the input parameters
+ * @param {object} config - config object derived from input JSON
+ * @param {Array} canvasSVG - d3 object of canvas svg
+ * @returns {undefined} - returns nothing
+ */
+const createEventline = (scale, config, canvasSVG) => {
+    const eventlineContent = canvasSVG
+        .append("g")
+        .classed(styles.eventlineContent, true)
+        .attr("clip-path", `url(#${config.clipPathId})`);
+
+    config.eventline.forEach((eventline, index) => {
+        const eventlineGroup = eventlineContent
+            .append("g")
+            .classed(styles.eventlineGroup, true)
+            .datum(eventline)
+            .attr("style", (val) => `fill: ${val.color}; stroke: ${val.color};`)
+            .attr("aria-selected", false);
+
+        eventlineGroup
+            .append("line")
+            .classed(styles.eventline, true)
+            .attr(
+                "style",
+                (d) =>
+                    `fill: ${d.color}; stroke: ${d.color}; stroke-dasharray: ${
+                        d.style !== undefined &&
+                        d.style.strokeDashArray !== undefined
+                            ? d.style.strokeDashArray
+                            : 0
+                    }`
+            )
+            .attr("pointer-events", "auto")
```

Comment:
Does this still work with `clickPassThrough`?

Reply:
Since we don't have onClick property I removed the clickPassThrough property.

Reply:
Right, the none of the elements below would be clickable. 
https://github.com/cerner/carbon-graphs/blob/master/docs/helpers/Dateline.md

Reply:
yes, none of the elements below would be clickable.

Reply:
Right, wont this be an issue if there is a data point beneath eventline?

Reply:
No, since the stroke width of the line is less then that of the data point we would be able to select the data point.

Reply:
How about in a mobile phone.

Reply:
In mobile view also I checked it won't be an issue.

Reply:
You need to keep in mind that there could be scenarios where there might be multiple eventlines.
In this case, 

![image](https://user-images.githubusercontent.com/4908886/71674058-eb2ce900-2d3f-11ea-9f1d-3a3a818aed24.png)

Reply:
since the z-index of the eventline is maximum so it will always overlap the datapoint, In such a scenario where eventlines are multiple is there a way to handle this?

because same persist in the dateline too.

Reply:
Which is why we have the clickPassThrough property.

Reply:
Should we add clickPassThrough property?

Reply:
Yes please.

Reply:
I have added clickPassThrough property.